### ISSUE TITLE
Padding=zero?

### ISSUE THREAD

**Issue by staxoverflowed** [iperov/DeepFaceLab#238]
I'm getting Padding=zero is not a valid padding type for conv2d? Have you implement your own zero padding? There is a keras.layers.ZeroPadding2D but I can't see where you are using it?

**Comment by staxoverflowed** [comment]
ah, my bad. Found it.

        class Conv2D():
            def __init__ (self, *args, **kwargs):
                self.reflect_pad = False
                padding = kwargs.get('padding','')
                if padding == 'zero':
                    kwargs['padding'] = 'same'
                if padding == 'reflect':
                    kernel_size = kwargs['kernel_size']
                    if (kernel_size % 2) == 1:
                        self.pad = (kernel_size // 2,)*2
                        kwargs['padding'] = 'valid'
                        self.reflect_pad = True
                self.func = keras.layers.Conv2D (*args, **kwargs)

            def __call__(self,x):
                if self.reflect_pad:
                    x = ReflectionPadding2D( self.pad ) (x)
                return self.func(x)
nnlib.Conv2D = Conv2D
